packages/react/polymorphic/src/forwardRefWithAs.ts

```diff
@@ -46,56 +52,14 @@ interface ForwardRefExoticComponentWithAs<
    * We don't use `React.ComponentType` here as we get type errors
    * when consumers try to do inline `as` components.
    */
-  <As extends React.ElementType<any>>(
-    props: MergeWithDOMProps<As, OwnProps & { as: As }>
-  ): JSX.Element;
-}
-
-/* -------------------------------------------------------------------------------------------------
- * forwardRefWithAs
- * -----------------------------------------------------------------------------------------------*/
-
-/**
```

Comment:
haha, nothing left in this file now 😀

I'm wondering though, do we still need the `never` thing? because how else would we remove native props? Let's catchup on that once I'm done with the review as I've left comments related to that elsewhere too.

Reply:
Haha yeah, wanted to chat to you about this one actually.